commands_test.go

```diff
@@ -6086,6 +6086,159 @@ var _ = Describe("Commands", func() {
+					   return 'Function 1'
+					end
+					
+					local function f2(keys, args)
+					   return 'Function 2'
+					end
+					
+					redis.register_function('other_func_1', f1)
+					redis.register_function{
+					 function_name='other_func_2',
+					 description ='This is the second function of lib 2',
+					 callback=f2,
+					 flags={'no-writes', 'allow-stale'}
+					}`
+		lib2Code = fmt.Sprintf(lib2Code, lib2Name)
+
+		//lib2Name := "mylib2"
+
+		AfterEach(func() {
+			flush := client.FunctionFlush(ctx)
+			Expect(flush.Err()).NotTo(HaveOccurred())
+		})
+
+		It("Loads a new library", func() {
+			functionLoad := client.FunctionLoad(ctx, lib1Code)
+			Expect(functionLoad.Err()).NotTo(HaveOccurred())
+
+			args := []interface{}{"function", "load", lib1Code}
+			Expect(functionLoad.Args()).To(Equal(args))
+
+			Expect(functionLoad.Val()).To(Equal(lib1Name))
+
+			functionList := client.FunctionList(ctx)
+			Expect(len(functionList.Val())).To(Equal(1))
+		})
+
+		It("Loads and replaces a new library", func() {
+			// Load a function for the first time
+			client.FunctionLoad(ctx, lib1Code)
+
+			// And then replace it
+			functionLoadReplace := client.FunctionLoadReplace(ctx, lib1Code)
+
+			Expect(functionLoadReplace.Err()).NotTo(HaveOccurred())
+
+			args := []interface{}{"function", "load", "replace", lib1Code}
+			Expect(functionLoadReplace.Args()).To(Equal(args))
+
+			Expect(functionLoadReplace.Val()).To(Equal(lib1Name))
+
+			functionList := client.FunctionList(ctx)
+			Expect(len(functionList.Val())).To(Equal(1))
+
+		})
+
+		It("Deletes a library", func() {
+			functionLoad := client.FunctionLoad(ctx, lib1Code)
+			Expect(functionLoad.Err()).NotTo(HaveOccurred())
+
+			functionDelete := client.FunctionDelete(ctx, lib1Name)
+			Expect(functionDelete.Err()).NotTo(HaveOccurred())
+
+			args := []interface{}{"function", "delete", lib1Name}
+			Expect(functionDelete.Args()).To(Equal(args))
+
+			functionList := client.FunctionList(ctx)
+			Expect(len(functionList.Val())).To(Equal(0))
+		})
+
+		It("Flushes all libraries", func() {
+			functionLoad1 := client.FunctionLoad(ctx, lib1Code)
+			Expect(functionLoad1.Err()).NotTo(HaveOccurred())
+
+			functionLoad2 := client.FunctionLoad(ctx, lib2Code)
+			Expect(functionLoad2.Err()).NotTo(HaveOccurred())
+
+			functionFlush := client.FunctionFlush(ctx)
+			Expect(functionFlush.Err()).NotTo(HaveOccurred())
+
+			args := []interface{}{"function", "flush"}
+			Expect(functionFlush.Args()).To(Equal(args))
+
+			functionList := client.FunctionList(ctx)
+			Expect(len(functionList.Val())).To(Equal(0))
+		})
+
+		It("Flushes all libraries asynchronously", func() {
+			FunctionLoad := client.FunctionLoad(ctx, lib1Code)
+			Expect(FunctionLoad.Err()).NotTo(HaveOccurred())
+
+			functionFlush := client.FunctionFlushAsync(ctx)
+			Expect(functionFlush.Err()).NotTo(HaveOccurred())
+
+			args := []interface{}{"function", "flush", "async"}
+			Expect(functionFlush.Args()).To(Equal(args))
```

Comment:
If someone has a better idea of testing specifically the `async` flushing, I'd love to hear it. Otherwise, I think that just testing the command string and the end result (no functions on the server) should be good enough.

Reply:
@elena-kolevska To specifically test the asynchronous behavior we can use a goroutine to perform some blocking operation, wait for a few seconds, and then call FunctionFlushAsync. We can then verify that the FunctionFlushAsync method does not block and returns without an error. Is that the right way to think?

Reply:
Yeah, great idea. Do you think we should add that to the test site, or would that fall more into the "testing the server" category?
I'm kind of inclined to think that for the commands that return a simple `OK` response where we don't need to populate a more complex object, checking that we've sent the correct string and we've gotten an OK should be good enough. What do you say?